hooks/cascading-scans/hook/hook.ts

```diff
@@ -216,29 +277,74 @@ function mergeCascadingRuleWithScan(
   // specified an empty list or map.
   let selectedTolerations = undefined;
   if (tolerations !== undefined) {
-    selectedTolerations = mergeInheritedArray(scan.spec.tolerations, tolerations, inheritTolerations);
+    selectedTolerations = mergeInheritedArray(
+      scan.spec.tolerations,
+      tolerations,
+      inheritTolerations
+    );
   } else if (inheritTolerations) {
     selectedTolerations = scan.spec.tolerations;
   }
 
   let selectedAffinity = undefined;
   if (affinity !== undefined) {
-    selectedAffinity = affinity
+    selectedAffinity = affinity;
   } else if (inheritAffinity) {
     selectedAffinity = scan.spec.affinity;
   }
-  
+
   return {
-    annotations: mergeInheritedMap(scan.metadata.annotations, scanAnnotations, inheritAnnotations),
+    annotations: mergeInheritedMap(
+      scan.metadata.annotations,
+      scanAnnotations,
+      inheritAnnotations
+    ),
     labels: mergeInheritedMap(scan.metadata.labels, scanLabels, inheritLabels),
     env: mergeInheritedArray(scan.spec.env, env, inheritEnv),
     volumes: mergeInheritedArray(scan.spec.volumes, volumes, inheritVolumes),
-    volumeMounts: mergeInheritedArray(scan.spec.volumeMounts, volumeMounts, inheritVolumes),
-    initContainers: mergeInheritedArray(scan.spec.initContainers, initContainers, inheritInitContainers),
-    hookSelector: mergeInheritedSelector(scan.spec.hookSelector, hookSelector, inheritHookSelector),
+    volumeMounts: mergeInheritedArray(
+      scan.spec.volumeMounts,
+      volumeMounts,
+      inheritVolumes
+    ),
+    initContainers: mergeInheritedArray(
+      scan.spec.initContainers,
+      initContainers,
+      inheritInitContainers
+    ),
+    hookSelector: mergeInheritedSelector(
+      scan.spec.hookSelector,
+      hookSelector,
+      inheritHookSelector
+    ),
     affinity: selectedAffinity,
-    tolerations: selectedTolerations
+    tolerations: selectedTolerations,
+  };
+}
+
+function hostOrIP(finding: Finding): string {
+  if (finding.attributes["hostname"]) {
+    return finding.attributes["hostname"];
+  }
+  if (finding.attributes["ip_address"]) {
+    return finding.attributes["ip_address"];
   }
+  if (finding.attributes["ip"]) {
+    return finding.attributes["ip"];
+  }
+  if (
+    finding.attributes["ip_addresses"] &&
+    isArray(
+      finding.attributes["ip_addresses"] &&
+        finding.attributes["ip_addresses"].length > 0
+    )
+  ) {
+    return finding.attributes["ip_addresses"][0];
```

Comment:
A comment referencing https://github.com/secureCodeBox/secureCodeBox/pull/1679 here would be nice. Because it seems like we're prioritizing the first IP address.

Reply:
That is because we do :D
The helper needs to pick out one address. I changed it so that it always consistently picks out the first address sorted alphabetically so that the address is consistent and doesn't depend on the order of ips in the field